File: user/update_links.py
import os
import re

dir_path = r'd:\pa teguh\frontend\user'

mappings = [
    ('Beranda', 'branda pmb.html'),
    ('Biodata', 'biodaata.html'),
    ('Formulir Pendaftaran', 'formulir pendaftaran.html'),
    ('Dokumen', 'dokumen step 1.html'),
    ('Status Pendaftaran', 'a_status.html'),
    ('Dashboard Akademik', 'dashboard akademik user.html'),
    ('KRS', 'krs.html'),
    ('KHS \(Nilai\)', 'khs user.html'),
    ('Transkrip', 'a_transkip_user.html'),
    ('Jadwal Kuliah', 'jadwal.html'),
    ('Pengumuman', 'informasi.html'),
    ('Diskusi / Forum', 'forum.html'),
    ('Keuangan', 'keuangan.html'),
    ('Keluar', 'login.html'),
]

for filename in os.listdir(dir_path):
    if filename.endswith('.html'):
        file_path = os.path.join(dir_path, filename)
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        for text, new_href in mappings:
            pattern = re.compile(rf'(<a[^>]*href=")([^"]*)("[^>]*>(?:(?!</?a>)[\s\S])*?){text}((?:(?!</?a>)[\s\S])*?</a>)', re.IGNORECASE)
            replacement_text = text.replace(r"\(", "(").replace(r"\)", ")")
            content = pattern.sub(rf'\g<1>{new_href}\g<3>{replacement_text}\g<4>', content)
            
            # Also handle btn-logout which is a button, wait Keluar might be a button!
            if text == 'Keluar':
                pattern_btn = re.compile(rf'(<button[^>]*class="[^"]*logout[^"]*"[^>]*>(?:(?!</?button>)[\s\S])*?){text}((?:(?!</?button>)[\s\S])*?</button>)', re.IGNORECASE)
                # For buttons we might want to change them to anchor tags?
                # The user just said "pastikan saling terhubung", if Keluar is a button, maybe wrap it in an <a> or add onclick.
                # Actually, in biodaata.html:
                # <button class="btn-logout">...Keluar</button>
                # Let's change button to anchor tag with href for Keluar.
                pass
            
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f'Updated {filename}')

File: user/update_links.js
const fs = require('fs');
const path = require('path');

const dir = 'd:\\pa teguh\\frontend\\user';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html'));

const mappings = [
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-home"[^>]*><\/i>\s*Beranda\s*<\/a>/gi, href: 'branda pmb.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-user"[^>]*><\/i>\s*Biodata\s*<\/a>/gi, href: 'biodaata.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-file-alt"[^>]*><\/i>\s*Formulir Pendaftaran\s*<\/a>/gi, href: 'formulir pendaftaran.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-folder"[^>]*><\/i>\s*Dokumen\s*<\/a>/gi, href: 'dokumen step 1.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-check-circle"[^>]*><\/i>\s*Status Pendaftaran\s*<\/a>/gi, href: 'a_status.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-columns"[^>]*><\/i>\s*Dashboard Akademik\s*<\/a>/gi, href: 'dashboard akademik user.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-file-signature"[^>]*><\/i>\s*KRS\s*<\/a>/gi, href: 'krs.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-star"[^>]*><\/i>\s*KHS \(Nilai\)\s*<\/a>/gi, href: 'khs user.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-file-invoice"[^>]*><\/i>\s*Transkrip\s*<\/a>/gi, href: 'a_transkip_user.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-calendar-alt"[^>]*><\/i>\s*Jadwal Kuliah\s*<\/a>/gi, href: 'jadwal.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-bullhorn"[^>]*><\/i>\s*Pengumuman\s*<\/a>/gi, href: 'informasi.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-comments"[^>]*><\/i>\s*Diskusi \/ Forum\s*<\/a>/gi, href: 'forum.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-wallet"[^>]*><\/i>\s*Keuangan\s*<\/a>/gi, href: 'keuangan.html' },
    { regex: /<a[^>]*href="([^"]*)"[^>]*>(\s*)<i[^>]*class="[^"]*fa-sign-out-alt"[^>]*><\/i>\s*Keluar\s*<\/a>/gi, href: 'login.html' },
];

for (const file of files) {
    const filePath = path.join(dir, file);
    let content = fs.readFileSync(filePath, 'utf8');
    let changed = false;
    
    for (const mapping of mappings) {
        const newContent = content.replace(mapping.regex, (match) => {
            return match.replace(/href="[^"]*"/, `href="${mapping.href}"`);
        });
        if (newContent !== content) {
            content = newContent;
            changed = true;
        }
    }
    
    if (changed) {
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Updated ${file}`);
    }
}
